arangod/IResearch/IResearchRocksDBLink.cpp

```diff
@@ -119,16 +118,39 @@ IResearchRocksDBLink::IResearchRocksDBLink(
   _sparse = true;   // always sparse
 }
 
-IResearchRocksDBLink::~IResearchRocksDBLink() {
-  // NOOP
-}
-
 /*static*/ arangodb::IndexTypeFactory const& IResearchRocksDBLink::factory() {
   static const IndexFactory factory;
 
   return factory;
 }
 
+arangodb::Result IResearchRocksDBLink::drop() {
+  auto* engine = arangodb::EngineSelectorFeature::ENGINE;
+
+  if (!engine || !engine->inRecovery()) {
+    auto view = IResearchLink::view();
+    auto logValue = RocksDBLogValue::IResearchLinkDrop(
+      arangodb::Index::_collection.vocbase().id(),
+      arangodb::Index::_collection.id(),
+      view ? view->id() : 0, // 0 == invalid TRI_voc_cid_t according to transaction::Methods
```

Comment:
whats the point of writing a drop marker with invalid information ?

Reply:
that's just how it was written originally, i.e. this is code moved from another place in the file
IResearchRocksDBRecoveryHelper.cpp:213 dropCollectionFromView(...) doesn't even use it